sentry-core/src/error.rs

```diff
@@ -0,0 +1,61 @@
+use std::error::Error;
+
+use crate::protocol::{Event, Exception, Level};
+use crate::utils::parse_type_name_from_debug;
+
+/// Create a sentry `Event` from a `std::error::Error`.
+///
+/// A chain of errors will be resolved as well, and sorted oldest to newest, as
+/// described on https://develop.sentry.dev/sdk/event-payloads/exception/.
+///
+/// # Examples
+///
+/// ```
+/// use thiserror::Error;
+///
+/// #[derive(Debug, Error)]
+/// #[error("inner")]
+/// struct InnerError;
+///
+/// #[derive(Debug, Error)]
+/// #[error("outer")]
+/// struct OuterError(#[from] InnerError);
+///
+/// let event = sentry_core::event_from_error(&OuterError(InnerError));
+/// assert_eq!(event.level, sentry_core::protocol::Level::Error);
+/// assert_eq!(event.exception.len(), 2);
+/// assert_eq!(&event.exception[0].ty, "InnerError");
+/// assert_eq!(event.exception[0].value, Some("inner".into()));
+/// assert_eq!(&event.exception[1].ty, "OuterError");
+/// assert_eq!(event.exception[1].value, Some("outer".into()));
+/// ```
+pub fn event_from_error<E: Error + ?Sized>(err: &E) -> Event<'static> {
+    let mut exceptions = vec![exception_from_error(err)];
+
+    let mut source = err.source();
+    while let Some(err) = source {
+        exceptions.push(exception_from_error(err));
+        source = err.source();
+    }
+
+    exceptions.reverse();
+    Event {
+        exception: exceptions.into(),
+        level: Level::Error,
+        ..Default::default()
```

Comment:
Can we capture a backtrace here on nightly?
See how [anyhow does it](https://github.com/dtolnay/anyhow/blob/2aa0cc1aba82a1ea513ba9e94d0077fd4d62479d/src/lib.rs#L201)

Reply:
not right now. I moved backtrace parsing out of core. I need to rethink how all this should fit together.

Reply:
Alright, this is an important point to reconsider, then. Eventually you will want to also add backtraces, so maybe this needs to be changed considering that it's a valid use case to capture errors just with `sentry_core`, and expect backtraces.